test/test_torch.py

```diff
@@ -4278,6 +4278,52 @@ def test_exponential_no_zero(self, device, dtype):
         x = torch.empty(50000000, device=device, dtype=dtype).exponential_()
         self.assertTrue(x.min() > 0)
 
+    @dtypes(torch.float, torch.cfloat)
+    def test_cov(self, device, dtype):
+        def check(t, correction=1, fweights=None, aweights=None):
+            actual = torch.cov(t, correction=correction, fweights=fweights, aweights=aweights)
+            t = t.cpu().numpy()
+            fweights = fweights.cpu().numpy() if fweights is not None else None
+            aweights = aweights.cpu().numpy() if aweights is not None else None
+            expected = np.cov(t, ddof=correction, fweights=fweights, aweights=aweights)
+            expected = torch.from_numpy(np.array(expected)).to(dtype=actual.dtype)
+            self.assertEqual(actual, expected, atol=1e-05, rtol=1e-05)
+
+        def generate_input_tensors():
+            yield torch.tensor(1, dtype=dtype, device=device)
+            yield make_tensor((1,), device, dtype)
+            yield make_tensor((10,), device, dtype)
+            yield make_tensor((5, 1), device, dtype)
```

Comment:
Does (5,1) input produce NaN output? It produces nan outputs in numpy if no other params are passed

Reply:
Good question, it does produce nan just as NumPy.